### PR TITLE
recover_cfg fix for arm

### PR DESCRIPTION
**Pull request by fr0zenrain**
i have try to recover_cfg for an arm binary,but i failed, and i just fix a little, but now the arm disassemble can't recognize the data in an arm function finally,so it will failed again.

### CODE REVIEW

File: barf/arch/arm/armbase.py [hunk 1]
```diff
@@ -263,7 +263,7 @@ def max_instruction_size(self):
         """
         instruction_size_map = {
             ARCH_ARM_MODE_ARM : 4,
-            ARCH_ARM_MODE_THUMB : 2,
+            ARCH_ARM_MODE_THUMB : 4,
```
Comment (cnheitman) [inline barf/arch/arm/armbase.py]: I am not sure why this change. As long as I know it the instruction size for thumb is 2 bytes.

File: barf/barf.py [hunk 2]
```diff
@@ -292,10 +292,17 @@ def recover_cfg(self, ea_start=None, ea_end=None, symbols=None, callback=None, a
         # Set architecture in case it wasn't already set.
         if arch_mode is None:
             arch_mode = self.binary.architecture_mode
-
+        else:
+            if arch_mode == arch.ARCH_ARM:
```
Comment (cnheitman) [inline barf/barf.py]: arch_mode can only be ARCH_ARM_MODE_ARM and ARCH_ARM_MODE_THUMB. Is this a typo?

Comment (fr0zenrain) [inline barf/barf.py]: thanks, i will change

File: barf/arch/arm/armbase.py (hunk 1 above)
Comment (fr0zenrain) [inline barf/arch/arm/armbase.py]: thum2 is 4 byte
